source/common/common/fancy_logger.cc

```diff
@@ -109,24 +141,87 @@ void FancyContext::initSink() {
   }
 }
 
-spdlog::logger* FancyContext::createLogger(std::string key, int level)
+spdlog::logger* FancyContext::createLogger(std::string key)
     ABSL_EXCLUSIVE_LOCKS_REQUIRED(fancy_log_lock_) {
   SpdLoggerSharedPtr new_logger =
       std::make_shared<spdlog::logger>(key, Logger::Registry::getSink());
   if (!Logger::Registry::getSink()->hasLock()) { // occurs in benchmark test
     initSink();
   }
-  level_enum lv = Logger::Context::getFancyDefaultLevel();
-  if (level > -1) {
-    lv = static_cast<level_enum>(level);
-  }
-  new_logger->set_level(lv);
+
+  new_logger->set_level(getLogLevel(key));
   new_logger->set_pattern(Logger::Context::getFancyLogFormat());
   new_logger->flush_on(level_enum::critical);
   fancy_log_map_->insert(std::make_pair(key, new_logger));
   return new_logger.get();
 }
 
+void FancyContext::updateVerbositySetting(std::vector<std::pair<absl::string_view, int>> updates) {
+  absl::WriterMutexLock ul(&fancy_log_lock_);
+  log_update_info_.clear();
+  for (const auto& [glob, level] : updates) {
+    if (level < 0 || level > 6) {
```

Comment:
these special numbers aren't that readable.

Also should these be saturated or simply ignored?